test/end-to-end/components/AvailableComponents.component.js

```diff
@@ -0,0 +1,37 @@
+// "Available Components" component in Edit Blueprint page
+class AvailableComponents {
```

Comment:
So the idea is that components/ contains abstraction classes for the things the tests want to handle, so that the logic to enumerate table elements, click things, etc. is centralized?

My gut feeling is that this will make the tests easier to maintain once you understand them, but harder to debug (as you can't easily follow the code when reading the selenium debug log) and understand when you aren't familiar with them. But I don't have practical experience with this, so this is just thinking aloud. If you prefer this style, then by all means.

Reply:
@martinpitt. I understand what you mean. I was thinking this a lot, I tried both ways(put it into page file or separate file). No one is the best. The only reason I choose this way is this's a React app, the test should follow React way.